pyproject.toml

```diff
@@ -60,8 +60,9 @@ test = [
   'setuptools >= 56.0.0; python_version >= "3.10"',
 ]
 typing = [
-  "importlib-metadata >= 4.6.4",
-  "mypy == 0.942",
+  "importlib-metadata >= 5.1",
+  "mypy == 0.991",
+  "tomli",
```

Comment:
Why do wee need tomli here?

Reply:
Because if you run the type check from 3.11, it won't have tomli and the type checking fails due to a missing library. Same reason we have importlib-metadata.

Reply:
I was confused because mypy should not type check code which is unreachable.  But then I remembered we are enforcing 3.7 in our `pyproject.toml`:

```toml
python_version = "3.7"
```

Not the first time it tripped me up -- this should go in tox, not in our config.

Reply:
This would mean mypy would likely be broken out of the box unless run though tox. The "current Python" default means we'd have to support type checking on all versions of Python, which is hard(er) or impossible (due to type ignores becoming invalid). We have designed our types (and type ignores) targeting 3.7, and if someone wants to target some other version, they can explicitly do it on the command line.

Reply:
I don't think that it's particularly hard or impossible (we are talking about two or three type ignores, all relating to importlib-metadata) and best practice would be to type check across all versions we support.  But we can tackle this in a separate PR.